File: cinema-ticket/src/main/kotlin/com/example/cinematicket/models/Calculator.kt
package domain.models

fun calc(movie: Movie, visitor: Visitor, priceList: PriceList, discount: Discount): Int {
    var price: Int = priceList.get(visitorType = visitor.visitorType)
    price -= discount.value
    price += if (movie.is3DTitle()) 400 else 0
    return price
}

File: CinemaTicket/src/domain/models/calculator.ts
import Movie from './movie';
import { Visitor } from './visitor';
import { PriceList } from './priceTable';
import Discount from './discount';

export default class Calculator {
  static calc(movie: Movie, visitor: Visitor, priceList: PriceList, discount: Discount): number {
    let price = priceList[visitor.visitorType];
    price -= discount.value;
    price += (movie.is3DTitle() ? 400 : 0);
    return price;
  }
}
